tests/e2e/spec/tonedear.e2e.spec.js

```diff
@@ -0,0 +1,220 @@
+
+/**
+ * Run the tests
+ * @param {WebDriver} driver Selenium WebDriver object
+ * @param {Array} modes Array of modes to run the tests in
+ * @param {boolean} keepDriver Whether to keep the driver open after the tests have run
+ * @returns {Promise<void>} A Promise for the completion of the tests
+ */
+function runTests (driver, modes, keepDriver) {
+    let browserName, browserVersion;
+    driver.getCapabilities().then(function (caps) {
+        browserName = caps.get('browserName');
+        browserVersion = caps.get('browserVersion');
+        console.log('\nRunning Tonedear tests on: ' + browserName + ' ' + browserVersion);
+    });
+
+    // Set the implicit wait to 3 seconds
+    driver.manage().setTimeouts({ implicit: 3000 });
+
+    // Run in both jquery and serviceworker modes by default
+    if (!modes) {
+        modes = ['jquery', 'serviceworker'];
+    }
+
+    modes.forEach(function (mode) {
+        let serviceWorkerAPI = true;
+        describe('Tonedear test ' + (mode === 'jquery' ? '[JQuery mode]' : '[SW mode]'), function () {
+            this.timeout(60000);
+            this.slow(10000);
+
+            it('Load Kiwix JS and check title', async function () {
+                await driver.get('http://localhost:' + port + '/dist/www/index.html?noPrompts=true');
+                await driver.sleep(1300);
+                await driver.navigate().refresh();
+                await driver.sleep(800);
+                const title = await driver.getTitle();
+                assert.equal('Kiwix', title);
+            });
+
+            it('Switch to ' + mode + ' mode', async function () {
+                const modeSelector = await driver.wait(until.elementLocated(By.id(mode + 'ModeRadio')));
+                await driver.wait(async function () {
+                    const elementIsVisible = await driver.executeScript(
+                        'var el=arguments[0]; el.scrollIntoView(true); setTimeout(function () {el.click();}, 50); return el.offsetParent;',
+                        modeSelector
+                    );
+                    return elementIsVisible;
+                }, 5000);
+                await driver.sleep(1300);
+
+                try {
+                    const activeAlertModal = await driver.findElement(By.css('.modal[style*="display: block"]'));
+                    if (activeAlertModal) {
+                        serviceWorkerAPI = await driver.findElement(By.id('modalLabel')).getText().then(function (alertText) {
+                            return !/ServiceWorker\sAPI\snot\savailable/i.test(alertText);
+                        });
+                        const approveButton = await driver.wait(until.elementLocated(By.id('approveConfirm')));
+                        await approveButton.click();
+                    }
+                } catch (e) {
+                    // Do nothing
+                }
```

Comment:
@THEBOSS0369 I think you could remove these lines. Although it's not the cause of the current error, it will never now find this element, and it introduces overhead (and confusion about the cause of failure in the logs). I'll test on a local copy of FF70 to see if there's an obvious cause of the failure to navigate to the Audio app page.